tests/src/test/scala/whisk/core/loadBalancer/test/LoadBalancerDataTests.scala

```diff
@@ -0,0 +1,135 @@
+    }
+
+    it should "return the number of activations for each invoker" in {
+
+        val loadBalancerData = new LoadBalancerData()
+        loadBalancerData.putActivation(firstEntry)
+        loadBalancerData.putActivation(secondEntry)
+
+        val result = loadBalancerData.activationCountByInvoker
+
+        result shouldBe Map(firstEntry.invokerName -> 1, secondEntry.invokerName -> 1)
+
+        loadBalancerData.activationCountByInvoker("invoker0") shouldBe 1
+        loadBalancerData.activationCountByInvoker("invoker1") shouldBe 1
+        loadBalancerData.activationById(firstEntry.id) shouldBe Some(firstEntry)
+        loadBalancerData.activationById(secondEntry.id) shouldBe Some(secondEntry)
+    }
+
+    it should "remove activations from all 3 maps" in {
+
+        val loadBalancerData = new LoadBalancerData()
+        loadBalancerData.putActivation(firstEntry)
+        loadBalancerData.putActivation(secondEntry)
+
+        loadBalancerData.removeActivation(firstEntry)
+        loadBalancerData.removeActivation(secondEntry)
+
+        val activationsByInvoker = loadBalancerData.activationCountByInvoker
+        val activationsByNamespace = loadBalancerData.activationCountByNamespace
+
+        activationsByInvoker.values.sum shouldBe 0
+        activationsByNamespace.values.sum shouldBe 0
+        loadBalancerData.activationById(firstEntry.id) shouldBe None
+        loadBalancerData.activationById(secondEntry.id) shouldBe None
+    }
+
+    it should "remove activations from all 3 maps by activation id" in {
+
+        val loadBalancerData = new LoadBalancerData()
+        loadBalancerData.putActivation(firstEntry)
+
+        loadBalancerData.removeActivation(firstEntry.id)
+
+        val activationsByInvoker = loadBalancerData.activationCountByInvoker
+        val activationsByNamespace = loadBalancerData.activationCountByNamespace
+
+        activationsByInvoker.values.sum shouldBe 0
+        activationsByNamespace.values.sum shouldBe 0
+        loadBalancerData.activationById(firstEntry.id) shouldBe None
+
+    }
+
+    it should "return None if the entry doesn't exist when we remove it" in {
+        val loadBalancerData = new LoadBalancerData()
+        loadBalancerData.removeActivation(firstEntry) shouldBe None
+    }
+
+    it should "respond with different values accordingly" in {
+
+        val entry = ActivationEntry(ActivationId(), UUID(), "invoker1", Instant
+          .now, activationIdPromise)
+        val entrySameInvokerAndNamespace = entry.copy(id = ActivationId())
+        val entrySameInvoker = entry.copy(id = ActivationId(), namespaceId = UUID())
+
+        val loadBalancerData = new LoadBalancerData()
+        loadBalancerData.putActivation(entry)
+        loadBalancerData.activationCountByNamespace(entry.namespaceId) shouldBe 1
+        loadBalancerData.activationCountByInvoker(entry.invokerName) shouldBe 1
+
+        loadBalancerData.putActivation(entrySameInvokerAndNamespace)
```

Comment:
It should perhaps be an error to put the same entry more than once. See my comments above.